src/layout/NestedPicker.js

```diff
@@ -37,19 +37,24 @@ FilteredSection.displayName = 'FilteredSection'
 
 let Section = observer(({ options, onClick, selected, Item }) => (
   <div>
-    {F.mapIndexed(
-      (item, key) => (
+    {_.map(
+      item => (
         <Item
-          key={key}
-          onClick={() => onClick(item.value || key, item)}
-          active={selected === key}
-          disabled={selected && selected !== key}
+          key={item._key}
+          onClick={() => onClick(item.value || item._key, item)}
+          active={selected === item._key}
+          disabled={selected && selected !== item._key}
           hasChildren={!isField(item)}
         >
-          {isField(item) ? item.shortLabel || item.label : _.startCase(key)}
+          {isField(item)
+            ? item.shortLabel || item.label
+            : _.startCase(item._key)}
         </Item>
       ),
-      options
+      _.flow(
+        F.unkeyBy('_key'),
+        _.sortBy(item => _.toLower(item._key))
```

Comment:
If we do this, we're only sorting by key - but if the labels are fundamentally different from the keys here, it seems like fields might appear to be in random orders, right? So we should probably still do the cascade from before